File: C/check_palindrome.c
#include <stdio.h>
#include <string.h>

int isPalindrome(char *str, int n) {
    int left = 0, right = n - 1;
    while (left < right) {
        if (str[left] != str[right])
            return 0;
        left++;
        right--;
    }
    return 1;
}

int main() {
    char str[100];
    printf("Enter a string: ");
    scanf("%s", str);
    int n = strlen(str);
    
    if (isPalindrome(str, n))
        printf("String is palindrome\n");
    else
        printf("String is not palindrome\n");
    return 0;
}

File: Python/check_palindrome.py
def is_palindrome(s, n):
    left, right = 0, n-1
    while left < right:
        if s[left] != s[right]:
            return False
        left += 1
        right -= 1
    return True

if __name__ == "__main__":
    s = input("Enter a string: ")
    n = len(s)
    result = "is" if is_palindrome(s, n) else "is not"
    print(f"String {result} a palindrome")